middle_end/flambda2/types/meet_and_join.ml

```diff
@@ -439,7 +436,8 @@ and meet_head_of_kind_naked_immediate env (t1 : TG.head_of_kind_naked_immediate)
     if I.Set.is_empty is
     then Bottom
     else
-      Ok (TG.Head_of_kind_naked_immediate.create_naked_immediates is, TEE.empty)
+      let<+ ty = TG.Head_of_kind_naked_immediate.create_naked_immediates is in
+      ty, TEE.empty
```

Comment:
You might as well remove the emptiness check above, it's redundant now.

Reply:
Removed